install/kubernetes/cilium/templates/cilium-operator-serviceaccount.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if .Values.operator.enabled }}
+{{- if and .Values.operator.enabled .Values.serviceAccounts.operator.create }}
```

Comment:
While this one fixes the reported github issue, I have one question which needs your inputs.

I am guessing the reason behind having the flag .Values.serviceAccounts.*.create is to allow end user avoid creating new SA, and use _existing_ SA. However, the current deployment manifest is having SA hard coded as per https://github.com/cilium/cilium/blob/master/install/kubernetes/cilium/templates/cilium-operator-deployment.yaml#L208. 

What do do think about having one simple logic like below ?

```
{{- if Values.serviceAccounts.*.create }}
serviceAccountName: <default SA> (e.g. cilium-operator)
{{- else }}
serviceAccountName: <SA from values.yaml) (e.g. .Values.operator.serviceAccountName)
{{- end }}
```

Reply:
Hey Tam, thanks for your feedback!
You are 100% right. I pushed some changes.